crypto/bio/bio_print.c

```diff
@@ -8,947 +8,71 @@
-        ++padlen;
-    }
-    return 1;
-}
-
-#endif /* OPENSSL_SYS_UEFI */
-
-#define BUFFER_INC  1024
-
-static int
-doapr_outch(char **sbuffer,
-            char **buffer, size_t *currlen, size_t *maxlen, int c)
-{
-    /* If we haven't at least one buffer, someone has done a big booboo */
-    if (!ossl_assert(*sbuffer != NULL || buffer != NULL))
-        return 0;
-
-    /* |currlen| must always be <= |*maxlen| */
-    if (!ossl_assert(*currlen <= *maxlen))
-        return 0;
-
-    if (buffer && *currlen == *maxlen) {
-        if (*maxlen > INT_MAX - BUFFER_INC)
-            return 0;
-
-        *maxlen += BUFFER_INC;
-        if (*buffer == NULL) {
-            if ((*buffer = OPENSSL_malloc(*maxlen)) == NULL)
-                return 0;
-            if (*currlen > 0) {
-                if (!ossl_assert(*sbuffer != NULL))
-                    return 0;
-                memcpy(*buffer, *sbuffer, *currlen);
-            }
-            *sbuffer = NULL;
-        } else {
-            char *tmpbuf;
-
-            tmpbuf = OPENSSL_realloc(*buffer, *maxlen);
-            if (tmpbuf == NULL)
-                return 0;
-            *buffer = tmpbuf;
-        }
-    }
-
-    if (*currlen < *maxlen) {
-        if (*sbuffer)
-            (*sbuffer)[(*currlen)++] = (char)c;
-        else
-            (*buffer)[(*currlen)++] = (char)c;
-    }
-
-    return 1;
-}
-
-/***************************************************************************/
-
 int BIO_printf(BIO *bio, const char *format, ...)
 {
     va_list args;
     int ret;
 
     va_start(args, format);
-
     ret = BIO_vprintf(bio, format, args);
-
     va_end(args);
     return ret;
 }
 
 int BIO_vprintf(BIO *bio, const char *format, va_list args)
 {
-    int ret;
-    size_t retlen;
-    char hugebuf[1024 * 2];     /* Was previously 10k, which is unreasonable
-                                 * in small-stack environments, like threads
-                                 * or DOS programs. */
-    char *hugebufp = hugebuf;
-    size_t hugebufsize = sizeof(hugebuf);
-    char *dynbuf = NULL;
-    int ignored;
-
-    dynbuf = NULL;
-    if (!_dopr(&hugebufp, &dynbuf, &hugebufsize, &retlen, &ignored, format,
-                args)) {
-        OPENSSL_free(dynbuf);
-        return -1;
-    }
-    if (dynbuf) {
-        ret = BIO_write(bio, dynbuf, (int)retlen);
-        OPENSSL_free(dynbuf);
-    } else {
-        ret = BIO_write(bio, hugebuf, (int)retlen);
-    }
-    return ret;
+    int ret, i = -1;
+    char buff[1024];
+    va_list copy;
+
+    /* Does it fit in the fixed buffer? */
+    va_copy(copy, args);
+    ret = vsnprintf(buff, sizeof(buff), format, args);
+    if (ret < (int)sizeof(buff))
+	i = BIO_write(bio, buff, ret);
+    else {
+	/* Allocate enough space and try again. */
+	int size = ret + 1;
+	char *p = OPENSSL_malloc(size);
+
+	if (p != NULL) {
+	    ret = vsnprintf(p, size, format, copy);
+	    if (ret < size)
+		i = BIO_write(bio, p, ret);
+	    OPENSSL_free(p);
+	}
```

Comment:
There's a bunch of tabs that snuck in here, but since you're going to move to a proper code formatter, I'm going to ignore this.

Reply:
I gotta figure out what went wrong, why `:set et` didn't work.

Reply:
What often bites me is that it's local to the buffer.

Reply:
I had it set for openssl not quictls :(. fixed.